tests/expr10.py

```diff
@@ -0,0 +1,13 @@
+def test_UnaryOp():
+    a: i32
+    a = +4
+    a = -500
+
+    f: f32
+    f = +1.0
+    f = -183745.534
+
+    b1: bool
+    b2: bool
+    b1 = True
+    b2 = not False
```

Comment:
Can you add a test here:
```
b3: bool
b3 = not b2
```
I think you'll get an assert error, because you assume that the operand is a `ConstantLogical`.

Instead, `not` must produce the `UnaryOp` ASR operation. In addition, if there is a compile time value, it should be assign to the `value` member. If there is no compile time value, the `value` should be `nullptr`.

Reply:
> I think you'll get an assert error because you assume that the operand is a `ConstantLogical`

It does produce an ASR node:
```bash
               (=
                  (Var 2 b3)
                  (UnaryOp
                     Not
                     (Var 2 b2)
                     (Logical 1 []) ()) ())]
```
But no value member added in `tmp` since we assume operand is `ConstantLogical`. So, no assert error.

Reply:
> Can you add a test here

Done.